autoload/vsnip/snippet.vim

```diff
@@ -504,6 +438,58 @@ function! s:Snippet.position_to_offset(position) abort
   return l:offset
 endfunction
 
+"
+" traverse.
+"
+function! s:Snippet.traverse(node, callback) abort
+  let l:state = {
+  \   'offset': 0,
+  \   'before_text': '',
+  \ }
+  let l:context = {
+  \   'depth': 0,
+  \   'parent': v:null,
+  \   'parents': [],
+  \ }
+  call s:traverse(a:node, a:callback, l:state, l:context)
+endfunction
+function! s:traverse(node, callback, state, context) abort
+  let l:text = ''
+  let l:length = 0
+  if a:node.type !=# 'snippet'
+    let l:text = a:node.text()
+    let l:length = strchars(l:text)
+    if a:callback({
+    \   'node': a:node,
+    \   'text': l:text,
+    \   'length': l:length,
+    \   'parent': a:context.parent,
+    \   'parents': a:context.parents,
+    \   'depth': a:context.depth,
+    \   'offset': a:state.offset,
+    \   'before_text': a:state.before_text,
+    \   'range': [a:state.offset, a:state.offset + l:length],
```

Comment:
I will add the `in_known_variable` for the context.

Basically, the known variable's children have no meaning.
But Currently vsnip traverse and calculate the length of it, it is a potential bug.